### ISSUE TITLE
Suggest a new feature

### ISSUE THREAD

**Issue by romaricpascal** [twbs/bootstrap#35945]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

It'd be great if Bootstrap was providing functions or mixins to manipulate that utilities hash a bit more easily. I'm thinking something like:

- `utilities-add-values($utility-name, $values)` for adding values to a specific utility
- `utilities-set-values($utility-name, $values)` for replacing the values of a utility completely
- `utilities-remove-values($utility-name, $values)` for removing targeted values of a utility
- `utilities-set-options($utility-name, $options)` to set specific options like `responsive`
- `utilities-add($new-utility-name, $options)` to add a new utility class
- `utilities-remove($utility-name)` to completely remove a utility

I'm thinking mixins would provide the best API from a usage point of view, similarly to [what sass-mq does for adding breakpoints](https://github.com/sass-mq/sass-mq/blob/7ebe25b1481e15ded53381aa3e3b26f35f65fbaf/_mq.scss#L252). allowing to only do:

```scss
@include utilities-add-values(...)
```

instead of having to assign the result of the function each time:

```scss
$utilities: utilities-add-values(...)
```

### Motivation and context

The [API for manipulating the `$utilities` map is well documented](https://getbootstrap.com/docs/5.0/utilities/api/#using-the-api), but it relies on the low level `map-merge`. This makes it a bit of a mouthful to write the operations, especially for adding new values to existing utilities.

**Comment by mdo** [comment]
Love the idea!

**Comment by romaricpascal** [comment]
I'll get started on a PR then 😄 Thinking a getter function for the values may also come handy (in case they need using in `calc`), but that may conflict with https://github.com/twbs/bootstrap/pull/34102/files 😬 

Testing-wise, how do you handle this in Bootstrap? I saw this issue https://github.com/twbs/bootstrap/issues/28496, but it's tagged help wanted. In any case, I can get coding the functions and add tests later on. 

(for what it's worth, I've had a good experience experimenting with [true](https://www.oddbird.net/true/docs/) for [testing SCSS mixins and functions](https://gitlab.com/romaricpascal/markup-engineering-tools/-/tree/master/scss/utils/__tests__), but that's likely a discussion to have on the other issue 😄)

**Comment by mdo** [comment]
Yeah, I still have to swing back to the `util()` mixin. I'd love to have that in the future as well obviously.

Regarding testing, two stalled out starts into that:

1. https://github.com/twbs/bootstrap/pull/34967
2. https://github.com/twbs/bootstrap/tree/gs/test-sass-unit-tests

I'm very interested in having CSS tests, just haven't had time to dedicate to it.
